openssl_common.go

```diff
@@ -0,0 +1,98 @@
+// +build openssl
+
+package crypto
+
+import (
+	pb "github.com/libp2p/go-libp2p-crypto/pb"
+
+	openssl "github.com/spacemonkeygo/openssl"
+)
+
+// define these as separate types so we can add more key types later and reuse
+// code.
```

Comment:
Unfortunately, RSA is the only place where we use the PKCS1v15 signature format, as far as I can tell and I can't find an easy way to get something else from this openssl library.